src/components/GasFeeWarning.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { useEffect } from 'react'
+import { useTranslation } from 'react-i18next'
+import { StyleSheet } from 'react-native'
+import AppAnalytics from 'src/analytics/AppAnalytics'
+import { AppEvents } from 'src/analytics/Events'
+import InLineNotification, { NotificationVariant } from 'src/components/InLineNotification'
+import { Spacing } from 'src/styles/styles'
+import { PreparedTransactionsResult } from 'src/viem/prepareTransactions'
+
+export enum GasFeeWarningFlow {
+  Send = 'Send',
+  Swap = 'Swap',
+  Deposit = 'Deposit',
+  Withdraw = 'Withdraw',
+  Dapp = 'Dapp',
+}
+
+function GasFeeWarning({
+  prepareTransactionsResult,
+  flow,
+  onPressCta,
+  testIdPrefix,
+}: {
+  prepareTransactionsResult?: PreparedTransactionsResult
+  flow: GasFeeWarningFlow
+  onPressCta?: () => void
+  testIdPrefix?: string
+}) {
+  const { t } = useTranslation()
+
+  useEffect(() => {
+    if (prepareTransactionsResult && prepareTransactionsResult.type !== 'possible') {
+      AppAnalytics.track(AppEvents.show_gas_fee_warning, {
+        flow,
+        errorType: prepareTransactionsResult.type,
+        tokenId: feeCurrency.tokenId,
+      })
+    }
+  }, [prepareTransactionsResult])
+
+  if (!prepareTransactionsResult || prepareTransactionsResult.type === 'possible') {
+    return null
+  }
```

Comment:
If you these, you can remove the if statement within the `useEffect`.
```suggestion
  if (!prepareTransactionsResult || prepareTransactionsResult.type === 'possible') {
    return null
  }

  useEffect(() => {
    AppAnalytics.track(AppEvents.show_gas_fee_warning, {
      flow,
      errorType: prepareTransactionsResult.type,
      tokenId: feeCurrency.tokenId,
    })
  }, [prepareTransactionsResult])
```

Reply:
I tried that but got the linter error `React Hook "useEffect" is called conditionally.`, so I think it has to be after and have the `if` check